app/src/org/commcare/views/widgets/ComboboxWidget.java

```diff
@@ -0,0 +1,123 @@
+
+    public ComboboxWidget(Context context, FormEntryPrompt prompt, boolean permissive) {
+        super(context, prompt);
+        initChoices(prompt);
+        comboBox = new Combobox(context, choiceTexts, true, permissive, mQuestionFontSize);
+        addView(comboBox);
+
+        comboBox.setEnabled(!prompt.isReadOnly());
+        comboBox.setFocusable(!prompt.isReadOnly());
+        addListeners();
+        fillInPreviousAnswer(prompt);
+    }
+
+    private void initChoices(FormEntryPrompt prompt) {
+        choices = prompt.getSelectChoices();
+        choiceTexts = new Vector<>();
+        for (int i = 0; i < choices.size(); i++) {
+            choiceTexts.add(prompt.getSelectChoiceText(choices.get(i)));
+        }
+    }
+
+    private void addListeners() {
+        comboBox.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                widgetEntryChanged();
+            }
+        });
+
+        comboBox.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                clearWarningMessage();
+            }
+        });
+    }
+
+    private void fillInPreviousAnswer(FormEntryPrompt prompt) {
+        if (prompt.getAnswerValue() != null) {
+            String previousAnswerValue = ((Selection)prompt.getAnswerValue().getValue()).getValue();
+            for (int i = 0; i < choices.size(); i++) {
+                if (choices.get(i).getValue().equals(previousAnswerValue)) {
+                    comboBox.setText(choiceTexts.get(i));
+                    // setting this text will cause the dropdown to open up, which we don't want
+                    comboBox.dismissDropDown();
+                    break;
+                }
+            }
+        }
+    }
+
+    @Override
+    public IAnswerData getAnswer() {
+        // So that we can see any error message that gets shown as a result of this
+        comboBox.dismissDropDown();
+
+        String enteredText = comboBox.getText().toString();
+        if (choiceTexts.contains(enteredText)) {
+            int i = choiceTexts.indexOf(enteredText);
+            return new SelectOneData(new Selection(choices.elementAt(i)));
+        } else if ("".equals(enteredText)) {
+            return null;
+        } else {
+            return new InvalidData("The text entered is not a valid answer choice",
+                    new SelectOneData(new Selection(enteredText)));
+        }
+    }
+
+    @Override
+    public void clearAnswer() {
+        comboBox.setText("");
+    }
+
+    @Override
+    public void setFocus(Context context) {
```

Comment:
There's actually an important reason I didn't do this, but it's surprisingly convoluted to explain in writing. I'll come explain in person when you have a chance